src/components/Users/Users.js

```diff
@@ -42,37 +42,17 @@ const Users = (props) => {
     );
   };
 
-  const goUnfollow = (user) => {
-    toggleFollowingProgress(true, user.id);
-    axios.delete(`${MAIN_URL}follow/${user.id}`, {
-      withCredentials: true,
-      headers: {
-        'API-KEY': API_KEY,
-      },
-    })
-      .then((response) => {
-        if (response.data.resultCode === 0) {
-          props.unfollow(user.id);
-        }
-        toggleFollowingProgress(false, user.id);
-      });
-  };
+  const followBtn = (user) => (
+    <button type="submit" disabled={followingInProgress.some((id) => id === user.id)} onClick={() => follow(user.id)}>
+                Follow
+    </button>
+  );
 
-  const goFollow = (user) => {
-    toggleFollowingProgress(true, user.id);
-    axios.post(`${MAIN_URL}follow/${user.id}`, {}, {
-      withCredentials: true,
-      headers: {
-        'API-KEY': API_KEY,
-      },
-    })
-      .then((response) => {
-        if (response.data.resultCode === 0) {
-          props.follow(user.id);
-        }
-        toggleFollowingProgress(false, user.id);
-      });
-  };
+  const unfollowBtn = (user) => (
```

Comment:
I did not mean you should create two separate functions for these buttons.
I meant one function that returns a button with different text (Follow/Unfollow) and click handler (follow/unfollow) that you pass as an argument for the function.

Reply:
/done